src/backend/distributed/master/master_metadata_utility.c

```diff
@@ -772,6 +772,44 @@ BuildShardPlacementList(ShardInterval *shardInterval)
 	return shardPlacementList;
 }
 
+List *
+BuildShardPlacementListForGroup(int32 groupId)
```

Comment:
Needs a function comment (and two empty lines between functions)

Reply:
To be more descriptive, maybe `LoadAllShardPlacementsOnNodeGroup`.

Reply:
My understanding is that Load* functions are related to preparing cached data structures, which we don't do here?

Reply:
fair point, but I'd still like the `AllShardPlacementsOnNodeGroup` in there. Maybe even without a verb? we do that sometimes (e.g. `ShardPlacementList`)

Reply:
done now!